src/WebJobs.Script.WebHost/Security/KeyManagement/ISecretsRepositoryFactory.cs

```diff
@@ -2,13 +2,14 @@
 // Licensed under the MIT License. See License.txt in the project root for license information.
 
 using System;
+using Microsoft.Azure.WebJobs.Host;
 using Microsoft.Azure.WebJobs.Script.Config;
 using Microsoft.Extensions.Logging;
 
 namespace Microsoft.Azure.WebJobs.Script.WebHost
 {
     public interface ISecretsRepositoryFactory
     {
-        ISecretsRepository Create(ScriptSettingsManager settingsManager, WebHostSettings webHostSettings, ScriptHostConfiguration config, ILogger logger);
```

Comment:
This and the factory interface above - at what point does this become a general IFactory<T> interface, since at this point the interfaces just return a T?

Reply:
A lot of these are being removed as they are no longer needed.